src/main/scala/mesosphere/marathon/api/akkahttp/v2/GroupsController.scala

```diff
@@ -3,29 +3,53 @@ package api.akkahttp
 package v2
 
 import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes
 import akka.http.scaladsl.server.{ Directive1, Route }
-import mesosphere.marathon.api.v2.{ AppHelpers, PodsResource }
+import akka.stream.Materializer
+import mesosphere.marathon.api.GroupApiService
+import mesosphere.marathon.api.akkahttp.PathMatchers.GroupPathIdLike
+import mesosphere.marathon.api.akkahttp.Rejections.{ EntityNotFound, Message }
+import mesosphere.marathon.api.v2.{ AppHelpers, AppNormalization, PodsResource }
 import mesosphere.marathon.core.appinfo.{ AppInfo, GroupInfo, GroupInfoService, Selector }
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.core.group.GroupManager
 import mesosphere.marathon.plugin.auth.{ Authorizer, Identity, ViewGroup, Authenticator => MarathonAuthenticator }
-import mesosphere.marathon.state.{ Group, PathId }
+import mesosphere.marathon.raml.DeploymentResult
+import mesosphere.marathon.state.{ Group, PathId, Timestamp }
+import mesosphere.marathon.stream.Sink
 import play.api.libs.json.Json
-import mesosphere.marathon.api.akkahttp.PathMatchers.GroupPathIdLike
-import mesosphere.marathon.core.appinfo.GroupInfoService
-import mesosphere.marathon.core.election.ElectionService
 
-import scala.concurrent.ExecutionContext
+import scala.async.Async._
+import scala.concurrent.{ Await, Awaitable, ExecutionContext, Future }
+import scala.util.{ Failure, Success }
 
-class GroupsController(electionService: ElectionService, infoService: GroupInfoService)(
+class GroupsController(
+    electionService: ElectionService,
+    infoService: GroupInfoService,
+    groupManager: GroupManager,
+    groupApiService: GroupApiService,
+    val config: MarathonConf)(
     implicit
     val actorSystem: ActorSystem,
     val executionContext: ExecutionContext,
     val authenticator: MarathonAuthenticator,
-    val authorizer: Authorizer
+    val authorizer: Authorizer,
+    val materializer: Materializer
 ) extends Controller {
   import Directives._
   import mesosphere.marathon.api.akkahttp.EntityMarshallers._
   import mesosphere.marathon.api.v2.json.Formats._
 
+  private val forceParameter = parameter('force.as[Boolean].?(false))
+
+  /** convert app to canonical form */
+  private implicit val appNormalization: Normalization[raml.App] = {
```

Comment:
I'd go for an explicit normalization as in the pods controller. These implict normalizations are hard to track.